### ISSUE TITLE
Statistics for gateways and leafnodes

### ISSUE THREAD

**Issue by ripienaar** [nats-io/nats-server#1061]
- [x] Feature Request or Change Proposal
 
#### Use Case:

Monitoring of leaf node traffic and health
 
#### Proposed Change:

Expose leaf and gateway stats in the stats port. 

Specifically how many are connected (like current routes gauge), bytes across the wire. Connection counts - a number that increases with each connection to monitor frequent reconnects etc. I am sure lots more would make sense

#### Who Benefits From The Change(s)?
 
leaf and gateway users

#### Alternative Approaches

something to listen to the system publishes, but would prefer not to add a dependency

**Comment by ripienaar** [comment]
related https://github.com/nats-io/nats-server/issues/984

**Comment by kozlovic** [comment]
Working on the leafnodes, but gatewayz is already available (`http://localhost:8222/gatewayz`) since 2.0.0, but unfortunately the link was missing from the `http://localhost:8222` list of links. It has been corrected in 2.0.4. Could you check and see what's missing there?

**Comment by ripienaar** [comment]
https://github.com/nats-io/nats-server/blob/2959b982ea9f1eda59c837582bd26ac3805f8e22/server/monitor.go#L898-L915 suggests there are no statistics as such - bytes passed over a gateway connection etc, like we have for routes https://github.com/nats-io/nats-server/blob/2959b982ea9f1eda59c837582bd26ac3805f8e22/server/monitor.go#L551-L568

the message counts and sizes are what I refer to here

**Comment by kozlovic** [comment]
This is for /varz. You then have a dedicated /gatewayz page that then can list individual inbound/outbound connections. I am embedding ConnInfo so I get all the connection stats. Check the top-level [here](https://github.com/nats-io/nats-server/blob/2959b982ea9f1eda59c837582bd26ac3805f8e22/server/monitor.go#L1239) and [here](https://github.com/nats-io/nats-server/blob/2959b982ea9f1eda59c837582bd26ac3805f8e22/server/monitor.go#L1250) is the detail of a gateway connection.

**Comment by ripienaar** [comment]
yup, thats the snippets I already showed here :) no conninfo there, I take it you mean you'll be adding that yeah?

**Comment by kozlovic** [comment]
No it is not the same the links I showed points to the structure that is displayed on the /gatewayz page (the one you pointed is in section in /varz).
Here is specifically the inclusion of ConnInfo: https://github.com/nats-io/nats-server/blob/2959b982ea9f1eda59c837582bd26ac3805f8e22/server/monitor.go#L1252

**Comment by ripienaar** [comment]
Right. Doh. Need more sleep. Think it’s good then. Thank you.

**Comment by kozlovic** [comment]
I will keep this opened for the Leafnode.
